src/components/buoy-telemetry/BuoyMap.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <div class="mapboxgl-map-container">
+    <div class="date is-hidden-touch is-size-4">
+      <span>{{ formattedDate }}</span>
+    </div>
+    <div class="date-mobile is-hidden-desktop is-size-5">
+      <span>{{ formattedDate }}</span>
+    </div>
+    <div ref="el" class="mapbox-container" />
+  </div>
+
+  <!-- reference image for mapbox to pull from -->
+  <img
+    ref="imageEl"
+    :src="BuoyMarker"
+    style="
+      visibility: hidden;
+      height: 15px;
+      width: 15px;
+      position: absolute;
+      top: 0;
+    "
+  />
+</template>
+
+<script setup lang="ts">
+import { computed, inject, onMounted, ref, watch } from "vue";
+import mapboxgl from "mapbox-gl";
+import { scaleSqrt, scaleLinear } from "d3-scale";
+
+import BuoyMarker from "@/assets/illustrations/buoy-marker.svg";
+import { Data } from "../../utils/erddap";
+import { BuoyStore } from "../../store/buoy";
+
+interface Sample extends Data {
+  date: Date;
+}
+
+const store = inject("store") as BuoyStore;
+
+const props = defineProps<{
+  samples: Sample[];
+  formattedDate: string;
+  variable: string;
+}>();
+
+const annotatedSamples = computed(() => {
+  const domain = [0, 45];
+  const sqrtScale = scaleSqrt().domain(domain).range([0.2, 0.65]);
+  const colorScale = scaleLinear()
+    .domain(domain)
+    .range(["#93C572", "#32CD32", "#355E3B"]) // pistachio, lime green, hunter green
+    .clamp(true);
+  return props.samples
+    .filter((row) => row.variable === props.variable)
+    .map((row) => {
+      return {
+        ...row,
+        color: colorScale(row.value),
+        size: sqrtScale(row.value),
+      };
+    });
+});
+
+const el = ref<HTMLDivElement>(null);
+const imageEl = ref<HTMLImageElement>(null);
+const map = ref(null);
+onMounted(() => {
+  mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_ACCESS_TOKEN;
+
+  map.value = new mapboxgl.Map({
+    container: el.value,
+    style: "mapbox://styles/ccv-bot/ckmxra8oi0rsw17mzcbqrktzi",
+    center: [-71.35, 41.517],
```

Comment:
Could be good to have these coordinates also defined in a variable (or at least a comment for now of where this is)
I'm guessing it may also become reusable in future maps, so we should maybe start thinking of abstracting some of the shared configs - but maybe just an issue for now

Reply:
I added a comment; the centerpoint seems to change depending on which buoys are used